### PR TITLE
Added missing attributes to BGP neighbors peer group

### PR DESCRIPTION
**Pull request by stalabi1**
##### SUMMARY

I added the following OC attributes to the bgp_neighbors module for peer groups:

bfd: check-control-plane-failure, enabled, bfd-profile
disable-ebgp-connected-route-check
dont-negotiate-capability
ebgp-multihop : enabled, multihop-ttl
enforce-first-as
enforce-multihop
local address
local-as
local-as-no-prepend
local-as-replace-as
override-capability
passive-mode
pg-description
shutdown-message
solo-peer
strict-capability-match
timers: connect-retry
ttl-security-hops

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

sonic_bgp_neighbors

##### ADDITIONAL INFORMATION

[show_running_configuration_bgp_pg_output.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/8673691/show_running_configuration_bgp_pg_output.txt)

### CODE REVIEW

**Comment by kerry-meyer** [comment]
Testing Verification:

[regression-2022-05-11-13-07-45.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/8691399/regression-2022-05-11-13-07-45.pdf)

[bgp_neighbors_facts_UT (2).log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/8691402/bgp_neighbors_facts_UT.2.log)

[bgp_pg_output_log_summary.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/8691553/bgp_pg_output_log_summary.txt)

File: plugins/modules/sonic_bgp_neighbors.py [hunk 1]
```diff
@@ -121,6 +139,102 @@
...
+                type: str
+                required: True
+              encrypted:
+                description:
+                  - Indicates whether the password is encrypted text.
+                type: bool
+                default: False
+          pg_description:
+            description:
+              - A textual description of the interface.
```
Comment (kerry-meyer) [inline plugins/modules/sonic_bgp_neighbors.py:159]: Would it be more accurate to change the word "interface" to "peer group"?

File: plugins/modules/sonic_bgp_neighbors.py [hunk 2]
```diff
@@ -87,7 +87,20 @@
...
+                description:
+                  - Enables BFD liveliness check for a BGP peer.
+                type: bool
+              check_failure:
+                description:
+                  - Link dataplane status with control plane.
+                type: bool
+              profile:
+                description:
+                  - BFD Profile name format.
```
Comment (kerry-meyer) [inline plugins/modules/sonic_bgp_neighbors.py:102]: It looks like the word "format" at the end of this description should be deleted.

File: tests/regression/roles/sonic_bgp_neighbors/defaults/main.yml [hunk 3]
```diff
@@ -604,6 +699,28 @@ merged_tests:
...
       - bgp_as: "{{bgp_as_1}}"
+        peer_group:
+          - name: SPINE
+            dont_negotiate_capability: false
+            ebgp_multihop:
+               enabled: false
+               multihop_ttl: 2
+            enforce_first_as: false
+            enforce_multihop: false
+            local_address: 'Ethernet6'
```
Comment (kerry-meyer) [inline tests/regression/roles/sonic_bgp_neighbors/defaults/main.yml:710]: Are test cases needed for IPv4 and IPv6 local addresses?

File: tests/regression/roles/sonic_bgp_neighbors/defaults/main.yml [hunk 4]
```diff
@@ -564,6 +634,31 @@ merged_tests:
     state: merged
     input:
       - bgp_as: "{{bgp_as_1}}"
+        peer_group:
+          - name: SPINE
+            auth_pwd:
```
Comment (kerry-meyer) [inline tests/regression/roles/sonic_bgp_neighbors/defaults/main.yml:639]: Should a "modify" test case be added for the peer group "auth_pwd"?

Comment (stalabi1) [reply tests/regression/roles/sonic_bgp_neighbors/defaults/main.yml:639]: To modify, the auth_pwd has to be deleted first. The behavior is in the transformer code.

Comment (kerry-meyer) [reply tests/regression/roles/sonic_bgp_neighbors/defaults/main.yml:644]: Okay. Thanks for explaining. I will mark this conversation "resolved".

**Review by kerry-meyer — APPROVED** [review]
All of the current proposed changes and test results look good.

Approved.

**Pull request merged**